File: backend/check_area_coords.ts
import { createClient } from '@supabase/supabase-js';
import * as dotenv from 'dotenv';
import * as path from 'path';

dotenv.config({ path: path.join(__dirname, '.env') });

const cleanUrl = process.env.SUPABASE_URL!;
const cleanKey = process.env.SUPABASE_SERVICE_KEY!;

const supabase = createClient(cleanUrl, cleanKey);

async function main() {
  const { data, error } = await supabase
    .from('area_map_config')
    .select('area_number, google_map_url, coordinates')
    .in('area_number', ['⑨', '⑩', '⑭'])
    .order('area_number');

  if (error) {
    console.error('Error:', error);
    return;
  }

  for (const row of data || []) {
    console.log(`\nエリア: ${row.area_number}`);
    console.log(`Google Map URL: ${row.google_map_url}`);
    console.log(`座標: ${JSON.stringify(row.coordinates)}`);
    if (row.coordinates) {
      const c = typeof row.coordinates === 'string' ? JSON.parse(row.coordinates) : row.coordinates;
      if (c?.lat && c?.lng) {
        console.log(`Google Maps確認URL: https://www.google.com/maps?q=${c.lat},${c.lng}`);
      }
    }
  }
}

main();

File: backend/check_area_coords.py
import urllib.request
import json
import re

# .envから読み込む
env = {}
with open('.env', 'r', encoding='utf-8') as f:
    for line in f:
        line = line.strip()
        if '=' in line and not line.startswith('#'):
            k, v = line.split('=', 1)
            env[k.strip()] = v.strip().strip('"')

url = env.get('SUPABASE_URL', '')
key = env.get('SUPABASE_SERVICE_KEY', '')

print(f"URL: {url}")

# area_map_configから⑨⑩⑭を取得
import urllib.parse
areas = urllib.parse.quote('in.(⑨,⑩,⑭)')
api_url = f"{url}/rest/v1/area_map_config?area_number={areas}&select=area_number,google_map_url,coordinates&order=area_number"

req = urllib.request.Request(
    api_url,
    headers={
        'apikey': key,
        'Authorization': f'Bearer {key}',
        'Content-Type': 'application/json'
    }
)

with urllib.request.urlopen(req) as resp:
    data = json.loads(resp.read())

for row in data:
    print(f"\nエリア: {row['area_number']}")
    print(f"Google Map URL: {row.get('google_map_url', 'なし')}")
    coords = row.get('coordinates')
    if coords:
        if isinstance(coords, str):
            coords = json.loads(coords)
        lat = coords.get('lat')
        lng = coords.get('lng')
        print(f"座標: lat={lat}, lng={lng}")
        print(f"確認URL: https://www.google.com/maps?q={lat},{lng}")
    else:
        print("座標: なし")
